rest/tests/test_rest.py

```diff
@@ -0,0 +1,742 @@
+        )
+        new_term_slug = output_dict['slug']
+        self.assertEqual(input_dict['label'],
+                         Term.objects.get(slug=new_term_slug).label)
+        # never created a duplicate
+        self.assertEqual(1, Term.objects.count())
+
+        # test missing repository
+        self.get_term(self.repo.slug, vocab.slug, "xyz",
+                      expected_status=NOT_FOUND)
+
+        vocab2 = create_vocabulary(
+            self.repo.id,
+            "other name",
+            "description",
+            True,
+            Vocabulary.FREE_TAGGING,
+            1000
+        )
+
+        self.delete_term(self.repo.slug, vocab.slug, new_term_slug)
+
+        # make sure terms only show up under vocabulary they belong to
+        term1 = self.create_term(self.repo.slug, vocab.slug)
+        term2 = self.create_term(self.repo.slug, vocab2.slug)
+        self.create_term(self.repo.slug, vocab.slug,
+                         expected_status=BAD_REQUEST)
+
+        terms = self.get_terms(self.repo.slug, vocab.slug)
+        self.assertEqual(1, terms['count'])
+        terms = self.get_terms(self.repo.slug, vocab2.slug)
+        self.assertEqual(1, terms['count'])
+
+        self.get_term(self.repo.slug, vocab.slug, term1['slug'])
+        self.get_term(self.repo.slug, vocab.slug, term2['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term1['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term2['slug'])
+
+    def test_vocabulary_pagination(self):
+        """Test pagination for collections"""
+
+        expected = [
+            create_vocabulary(
+                self.repo.id,
+                "name{i}".format(i=i),
+                "description",
+                True,
+                Vocabulary.FREE_TAGGING,
+                1000
+            ) for i in range(100)]
+
+        resp = self.client.get(
+            '/api/v1/repositories/{repo_slug}/vocabularies/'.format(
+                repo_slug=self.repo.slug,
+            ))
+        self.assertEqual(OK, resp.status_code)
+        vocabularies = as_json(resp)
+        self.assertEqual(100, vocabularies['count'])
+        self.assertEqual([VocabularySerializer(x).data for x in expected[:20]],
+                         vocabularies['results'])
+
+        resp = self.client.get(
+            '/api/v1/repositories/{repo_slug}/vocabularies/?page=2'.format(
+                repo_slug=self.repo.slug,
+            ))
+        self.assertEqual(OK, resp.status_code)
+        vocabularies = as_json(resp)
+        self.assertEqual(100, vocabularies['count'])
+        self.assertEqual([VocabularySerializer(x).data
+                          for x in expected[20:40]], vocabularies['results'])
+
+    def test_authentication(self):
+        """Make sure unauthenticated and unauthorized users can't get access"""
+
+        anonymous = Client()
```

Comment:
LoreTestCase.logout/login deals with self.client but I want a different client logged in as a different user, or not logged in at all